lib/index.ts

```diff
@@ -96,7 +96,7 @@ class PWMetrics {
         console.log(messages.getMessage('MEDIAN_RUN'));
         this.displayOutput(results.median);
       } else if (this.flags.submit) {
-        const sheets = new Sheets(this.sheets, this.clientSecret);
+        const sheets = new Sheets(this.sheets, this.clientSecret, this.configIsSubmitAndUpload);
```

Comment:
What if not pass function to the sheets but extend `results.runs` with `viewerUrl` like
```
const sheets = new Sheets(this.sheets, this.clientSecret, this.configIsSubmitAndUpload);
await sheets.appendResults(extendResultsToSubmit(results.runs));

....
//extendResultsToSubmit method
return results.map(data => {
  if (this.flags.submit && this.flags.upload) {
    data.viewerUrl = getTimelineViewerUrl(data.fileId);
  }
  return data;
});
```
in this case we can cut off some stuff
wdyt?